src/Core/Content/ContactForm/SalesChannel/ContactFormRoute.php

```diff
@@ -135,10 +139,23 @@ public function load(RequestDataBag $data, SalesChannelContext $context): Contac
     private function validateContactForm(DataBag $data, SalesChannelContext $context): void
     {
         $definition = $this->contactFormValidationFactory->create($context);
+        $isPhoneOptionalInContactForm = $this->systemConfigService->get('core.basicInformation.isPhoneOptionalInContactForm', $context->getSalesChannel()->getId());
+
+        if ($isPhoneOptionalInContactForm === true) {
+            $definition->set('phone', new Optional());
+        }
```

Comment:
Hi @claudiobianco , would you mind solving this inside the `\Shopware\Core\Content\ContactForm\Validation\ContactFormValidationFactory`?